src/connection-manager/dial-queue.ts

```diff
@@ -253,18 +249,16 @@ export class DialQueue {
     return await pendingDial.promise
   }
 
-  private createDialAbortControllers (userSignal?: AbortSignal): { timeoutController: TimeoutController, signal: ClearableSignal } {
-    // ensure we throw if the dial takes longer than the dial timeout
-    const timeoutController = new TimeoutController(this.dialTimeout)
-
+  private createDialAbortControllers (userSignal?: AbortSignal): ClearableSignal {
     // let any signal abort the dial
-    const signal = combineSignals(
-      timeoutController.signal,
-      this.shutDownController.signal,
-      userSignal
+    const signal = anySignal(
```

Comment:
This signal needs to have `.clear` called on it after use.

Reply:
This particular signal is cleared properly in the [subsequent try catches](https://github.com/libp2p/js-libp2p/blob/2a9617b3e2fbc216c2596a2736643a6b68f6633f/src/connection-manager/dial-queue.ts#L232), adding it anywhere else would cause [this test](https://github.com/libp2p/js-libp2p/blob/2a9617b3e2fbc216c2596a2736643a6b68f6633f/test/connection-manager/dial-queue.spec.ts#L188) to fail.